lib/daru/dataframe.rb

```diff
@@ -2177,5 +2276,64 @@ def one_to_many_row row, number, vars, pattern
           [v, row[name]]
         }.to_h
     end
+
+    # Raises IndexError when one of the positions is a valid position
+    def validate_vector_positions *positions
+      positions = [positions] if positions.is_a? Integer
+      positions.each do |pos|
+        raise IndexError, "#{pos} is not a valid position." if pos >= shape[1]
```

Comment:
Also, I think in cases like this it is better to use `nrows`/`ncols` explicitly. `shape[x]` is not that readable, and it is pretty hard to spot bugs (e.g. rows and cols mixed up).